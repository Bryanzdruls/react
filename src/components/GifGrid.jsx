import React, { useEffect } from 'react'
import { getGifs } from '../helpers/GetGifs';

export const GifGrid = ({category}) => {
 
  useEffect(()=>{
    getGifs( category );
  },[])


  return (
    <>
        <h3> {category} </h3>        
    </>
  )
}
