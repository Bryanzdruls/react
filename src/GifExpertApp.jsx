import { useState } from 'react'
import { AddCategory, GifGrid } from './components'


export const GifExpertApp = () => {
  const [categories, setCategories] = useState([]);

  const onAddCategory = (NewCategory) => {
    if (categories.includes(NewCategory)) return;
    setCategories([NewCategory, ...categories]);
    //setCategories(  cat =>[...categories,'Valorant'] );
  }


  return (
    <>
      {/*Titulo  */}
      <h1>GifExpertApp</h1>

      {/* Input */}
      <AddCategory
        //onAddCategory={ setCategories }
        onNewCategory={(event) => onAddCategory(event)}
      />
      {/* Listado de tarjetas */}        
      {categories.map(cat => (
        <GifGrid 
          key={cat} 
          category={cat }
        />
      ))
      }
      {/* gif item */}
    </>
  )
}


