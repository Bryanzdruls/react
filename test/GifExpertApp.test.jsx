
import { fireEvent, render, screen } from "@testing-library/react"
import { GifExpertApp } from "../src/GifExpertApp"
import { AddCategory } from "../src/components";

describe('probando el componente <GifExpertApp/>', () => {
  const categories = ['naruto', 'bleach'];

   test('debe probar el onAddCategory debe retornar arreglo de categorias ', () => {        
        render( <GifExpertApp/>);
        //expect(container).toMatchSnapshot();
        screen.debug();
        const addCat =screen.getByRole('AddCategory');
        fireEvent.addCat(AddCategory, 'goku');

        console.log(categories);
        

    })
  
})
