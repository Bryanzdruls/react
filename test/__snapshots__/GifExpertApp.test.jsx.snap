// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`probando el componente <GifExpertApp/> debe probar el onAddCategory  1`] = `
<div>
  <h1>
    GifExpertApp
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Buscar Gifs"
      type="text"
      value=""
    />
  </form>
</div>
`;

exports[`probando el componente <GifExpertApp/> debe probar el onAddCategory debe retornar arreglo de categorias  1`] = `
<div>
  <h1>
    GifExpertApp
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Buscar Gifs"
      type="text"
      value=""
    />
  </form>
</div>
`;
